UnchainedLauncher.Core/Services/Processes/Chivalry/Chivalry2Launcher.cs

```diff
@@ -0,0 +1,36 @@
+﻿using LanguageExt;
+using log4net;
+using System.Diagnostics;
+using UnchainedLauncher.Core.Services.Processes.Chivalry.LaunchPreparers;
+using UnchainedLauncher.Core.Utilities;
+
+namespace UnchainedLauncher.Core.Services.Processes.Chivalry {
+    using static LanguageExt.Prelude;
+    public class Chivalry2Launcher : IChivalry2Launcher {
+        private static readonly ILog Logger = LogManager.GetLogger(typeof(Chivalry2Launcher));
+
+        private IChivalry2LaunchPreparer<ModdedLaunchOptions> LaunchPreparer { get; }
+        private IProcessLauncher Launcher { get; }
+        private string WorkingDirectory { get; }
+
+        public Chivalry2Launcher(
+            IChivalry2LaunchPreparer<ModdedLaunchOptions> preparer,
+            IProcessLauncher processLauncher,
+            string workingDirectory) {
+            WorkingDirectory = workingDirectory;
+            LaunchPreparer = preparer;
+            Launcher = processLauncher;
+        }
+
+        public Task<Either<LaunchFailed, Process>> Launch(string args, ModdedLaunchOptions options) {
```

Comment:
For a vanilla launch, what is used as `ModdedLaunchOptions`? At this point it seems like a misnomer if it's used in all launches